src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsClient.java

```diff
@@ -471,7 +472,10 @@ private byte[] doUdpQuery(Packet pkt, InetAddress server,
                             return cachedMsg;
                         }
                     }
-                    timeoutLeft = pktTimeout - ((int) (end - start));
+                    // Math.max ensures that the timeout is decreased
+                    long elapsedMillis = Math.max(1,
+                            TimeUnit.NANOSECONDS.toMillis(end - start));
+                    timeoutLeft = timeoutLeft - (int) elapsedMillis;
```

Comment:
Hello Aleksei, should this `int` cast take into account potential integer value overflow? In theory (and probably even in practice depending on what the initial timeout value was configured to), the `elapsedMillis`, I think can be a value greater than `Integer.MAX_VALUE`, in which case this cast to `int` can cause unexpected computation of `timeoutLeft`.

Reply:
Thanks for highlighting the overflow problem, Jaikiran. I agree that it could happen in practice when timeout and retries configuration properties specify timeout value close to `Integer.MAX_VALUE`. Addressed it in 30f883b8cb31120907002191dbfd88d787c75ec8.